frontend/src/data/variables/IncarcerationProvider.tsx

```diff
@@ -94,17 +97,9 @@ class IncarcerationProvider extends VariableProvider {
   ): Promise<MetricQueryResponse> {
     const breakdowns = metricQuery.breakdowns;
 
-    let dataType = "";
-    // determine JAIL vs PRISON based on the incoming requested metric ids
+    let variableId: VariableId | undefined = metricQuery?.variableId;
 
-    if (breakdowns.geography === "county") {
-      if (metricQuery.metricIds.some((id) => JAIL_METRICS.includes(id)))
-        dataType = "jail";
-      else if (metricQuery.metricIds.some((id) => PRISON_METRICS.includes(id)))
-        dataType = "prison";
-    }
-
-    const datasetId = this.getDatasetId(breakdowns, dataType);
```

Comment:
So instead of using this calculation to determine which dataset to retrieve, we're sending the variable id with the data set needed? Nice!!

Reply:
Thanks! Ya, all the original sources we were bundling all the datatype info into the same table (like with covid there are columns for hosp, for deaths and for cases). But now with Incarceration at the county level and for the CAWP over time stuff we need to store the datatypes in separate tables for various reasons. So i figured rather than do a ton of work to guess at what the calling VariableID was, we should just ship it along with the request :)